typescript/utils/src/typeof.ts

```diff
@@ -1,6 +1,7 @@
-export function isNullish<T>(val: T) {
-  if (val === null || val === undefined) return true;
-  else return false;
+export function isNullish<T>(
+  val: T,
+): val is T extends null | undefined ? T : never {
+  return val === null || val === undefined;
```

Comment:
the conditional type here is a bit hard to read

Reply:
Yeah complex types are like regex. Nearly unparseable by humans. I don't know of a simpler way to do this type narrowing though, sorry